platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/deduplication/EventDeduplicator.java

```diff
@@ -137,12 +138,13 @@ public GossipEvent handleEvent(@NonNull final GossipEvent event) {
     }
 
     /**
-     * Set the minimum generation required for an event to be non-ancient.
+     * Set the NonAncientEventWindow, defines the minimum threshold for an event to be non-ancient.
      *
-     * @param minimumGenerationNonAncient the minimum generation required for an event to be non-ancient
+     * @param nonAncientEventWindow the non-ancient event window
      */
-    public void setMinimumGenerationNonAncient(final long minimumGenerationNonAncient) {
-        this.minimumGenerationNonAncient = minimumGenerationNonAncient;
+    public void setNonAncientEventWindow(@NonNull final NonAncientEventWindow nonAncientEventWindow) {
+        // FUTURE WORK: change from minGenNonAncient to minRoundNonAncient
```

Comment:
same as my other comment, this probably needs to be updated in a lot of places

Reply:
updated